common/device-model.cpp

```diff
@@ -3349,21 +3349,40 @@ namespace rs2
         {
             if (sub->supports_on_chip_calib() && !has_autocalib)
             {
+                bool is_d555 = false;
+
+                if( dev.supports( RS2_CAMERA_INFO_PRODUCT_ID ) )
+                {
+                    auto pid_str = std::string( dev.get_info( RS2_CAMERA_INFO_PRODUCT_ID ) );
+                    if( pid_str == "0B56" || pid_str == "DDS" )
+                        is_d555 = true;
+                }
+
                 if (ImGui::Selectable("On-Chip Calibration", false, avoid_selection_flag))
                 {
                     try
                     {
-                        auto manager = std::make_shared<d500_on_chip_calib_manager>(viewer, sub, *this, dev);
-                        auto n = std::make_shared<d500_autocalib_notification_model>("", manager, false);
-                        viewer.not_model->add_notification(n);
+                        std::shared_ptr< process_manager > manager;
```

Comment:
d500_on_chip_calib_manager is related to TC on other d500 devices?
If yes it will need to be tested

Reply:
Yes, `d500_on_chip_calib_manager` is related also to TC. Both it and `on_chip_calib_manager` inherit from `process_manager`. I have used a pointer to the interface to allow greater flexibility on implementation being used.

`d500_on_chip_calib_manager` is still created in case it is not D555

Reply:
Did you test that SC TC was not impacted by these changes?

Reply:
No. Not expected to be affected by my changes as a different class instance is being created for D555 and there are no shared functions between UCAL and TC.

I will check anyway...